lighthouse-core/lib/i18n/swap-locale.js

```diff
@@ -65,21 +66,37 @@ function swapLocale(lhr, requestedLocale) {
         // `values` are the string template values to be used. eg. `values: {wastedBytes: 9028}`
         values = instance.values;
       }
-      // If we couldn't find the new replacement message, keep things as is.
-      try {
-        // Get new formatted strings in revised locale
-        const formattedStr = i18n.getFormattedFromIdAndValues(locale, icuMessageId, values);
-        // Write string back into the LHR
-        _set(lhr, path, formattedStr);
-      } catch (err) {
-        if (err.message === i18n._ICUMsgNotFoundMsg) {
-          missingIcuMessageIds.push(icuMessageId);
-        } else {
-          throw err;
+
+      // If the path isn't valid or the value isn't a string, there's no point in trying to replace it.
+      const originalString = _get(lhr, path);
+      if (typeof originalString !== 'string') {
+        // TODO(bckenny): what to do in this case? Ignore?
```

Comment:
@paulirish have we ever cared about `missingIcuMessageIds`? They're kind of helpful for tests, but I'm not sure if there's much value in expanding the kinds of errors we log to an array no one looks at :)

In this case the two TODOs are what to do with
- `icuMessagePath` that doesn't actually refer to a string in the LHR. This is either a bug in path generation (so should be caught by tests) or a malformed LHR (in which case...what could we even do), so ignoring may be fine?
- We log missing IDs, but what if the ID exists but the number or type of replacement values has changed? I think it would just throw on master, but do we want to put that case into `missingIcuMessageIds` as well?

Reply:
> @paulirish have we ever cared about missingIcuMessageIds? 

i don't. it came up while writing this but it's never been a problem in reality.